### ISSUE TITLE
[Egg Request]: HumanitZ

### ISSUE THREAD

**Issue by ValhallaCommunityOfficial** [parkervcp/eggs#2624]
### Does this expand an already existing service

No

### Link to a game or software

https://store.steampowered.com/app/1766060

### Links for the download

https://steamdb.info/app/2728330/depots/

### Links for the install docs

https://imgur.com/eWPkN1r

**Comment by ValhallaCommunityOfficial** [comment]
They posted an image of the server docs on discord so i uploaded it to imgur

**Comment by ImKringle** [comment]
What's your discord? I don't own the game so can't test that end if you would mind? I can send ya the ip

**Comment by ValhallaCommunityOfficial** [comment]
I think someone asked for a link to the humanitz discord then deleted it, here it is just incase https://discord.gg/GKAyNdJ2Fh

**Comment by ValhallaCommunityOfficial** [comment]
[humanitz.txt](https://github.com/parkervcp/eggs/files/13752099/humanitz.txt)

I have extracted the text from the doc image if that helps.

**Comment by vp-en** [comment]
I'm trying to make one. But if anyone's faster than me (I'm very new to creating eggs), they should probably go ahead and create it instead👍

**Comment by gOOvER** [comment]
> ### Does this expand an already existing service
> No
> 
> ### Link to a game or software
> https://store.steampowered.com/app/1766060
> 
> ### Links for the download
> https://steamdb.info/app/2728330/depots/
> 
> ### Links for the install docs
> https://imgur.com/eWPkN1r

a pic is not an install doc

**Comment by gOOvER** [comment]
> [humanitz.txt](https://github.com/parkervcp/eggs/files/13752099/humanitz.txt)
> 
> I have extracted the text from the doc image if that helps.

also not an install doc

**Comment by ValhallaCommunityOfficial** [comment]
> I'm trying to make one. But if anyone's faster than me (I'm very new to creating eggs), they should probably go ahead and create it instead👍

Please let me know if you get it working! :)

**Comment by vp-en** [comment]
Stuck for now. If anyone wants to continue the work I did, then you can try either one of these files:

**HumanitZ Server - v1**
This only includes the most basic variables, AppID, AutoUpdate, QueryPort and ServerName. Then trying to convert the included "[TSSGameServer.sh](https://paste.ec/paste/2KZLEXUa#Vt-oYB6fQDBuW3IRBiNv8Xk89O2a9FBzaIHOfmzLE1e)" to a usable start-up command in Pterodactyl

https://paste.ec/paste/YlPBn+oQ#aICECbERBzA8uABIxWxsXtyifGaf+vTbTfCckilUeYP


**HumanitZ Server - v2**
This includes all variables from their included "[ReadMe.txt](https://paste.ec/paste/MbXvpRMe#a2ofBY61M5S3lqdiUev8cfobRVtCwo4z9jPnQwTdxvm)". Not tested, since I can't get the server itself to launch properly.

https://paste.ec/paste/dmAv-1yz#TDN6AFGOEARz4QKNjspLb6jgIzXeobvqk9EMEDOB248

**Comment by ImKringle** [comment]
> Stuck for now, server is not launching properly. If anyone wants to continue the work I did, then you can try either one of these files. I don't need any credit or whatever, if you end up finishing/completing it.
> 
> **HumanitZ Server - v1** This only includes the most basic variables, AppID, AutoUpdate, QueryPort and ServerName. Then trying to convert the included "[TSSGameServer.sh](https://paste.ec/paste/2KZLEXUa#Vt-oYB6fQDBuW3IRBiNv8Xk89O2a9FBzaIHOfmzLE1e)" to a usable start-up command in Pterodactyl
> 
> https://paste.ec/paste/YlPBn+oQ#aICECbERBzA8uABIxWxsXtyifGaf+vTbTfCckilUeYP
> 
> **HumanitZ Server - v2** This includes all variables from their included "[ReadMe.txt](https://paste.ec/paste/MbXvpRMe#a2ofBY61M5S3lqdiUev8cfobRVtCwo4z9jPnQwTdxvm)". Not tested, since I can't get the server itself to launch properly.
> 
> https://paste.ec/paste/dmAv-1yz#TDN6AFGOEARz4QKNjspLb6jgIzXeobvqk9EMEDOB248

You hit the same block I did. It seems to hang with the SteamAPI check. Mine was using the sh file rather than the startup you have set
- 
[egg-humanit-z.json](https://github.com/parkervcp/eggs/files/13756271/egg-humanit-z.json)

**Comment by ValhallaCommunityOfficial** [comment]
> > Stuck for now, server is not launching properly. If anyone wants to continue the work I did, then you can try either one of these files. I don't need any credit or whatever, if you end up finishing/completing it.
> > **HumanitZ Server - v1** This only includes the most basic variables, AppID, AutoUpdate, QueryPort and ServerName. Then trying to convert the included "[TSSGameServer.sh](https://paste.ec/paste/2KZLEXUa#Vt-oYB6fQDBuW3IRBiNv8Xk89O2a9FBzaIHOfmzLE1e)" to a usable start-up command in Pterodactyl
> > https://paste.ec/paste/YlPBn+oQ#aICECbERBzA8uABIxWxsXtyifGaf+vTbTfCckilUeYP
> > **HumanitZ Server - v2** This includes all variables from their included "[ReadMe.txt](https://paste.ec/paste/MbXvpRMe#a2ofBY61M5S3lqdiUev8cfobRVtCwo4z9jPnQwTdxvm)". Not tested, since I can't get the server itself to launch properly.
> > https://paste.ec/paste/dmAv-1yz#TDN6AFGOEARz4QKNjspLb6jgIzXeobvqk9EMEDOB248
> 
> You hit the same block I did it seems lol. It seems to hang with the SteamAPI check. Mine was using the sh file rather than the startup you have set [egg-humanit-z.json](https://github.com/parkervcp/eggs/files/13756281/egg-humanit-z.json)

What information would be needed to get past this roadblock? I can ask on their discord.

**Comment by ImKringle** [comment]
> > > Stuck for now, server is not launching properly. If anyone wants to continue the work I did, then you can try either one of these files. I don't need any credit or whatever, if you end up finishing/completing it.
> > > **HumanitZ Server - v1** This only includes the most basic variables, AppID, AutoUpdate, QueryPort and ServerName. Then trying to convert the included "[TSSGameServer.sh](https://paste.ec/paste/2KZLEXUa#Vt-oYB6fQDBuW3IRBiNv8Xk89O2a9FBzaIHOfmzLE1e)" to a usable start-up command in Pterodactyl
> > > https://paste.ec/paste/YlPBn+oQ#aICECbERBzA8uABIxWxsXtyifGaf+vTbTfCckilUeYP
> > > **HumanitZ Server - v2** This includes all variables from their included "[ReadMe.txt](https://paste.ec/paste/MbXvpRMe#a2ofBY61M5S3lqdiUev8cfobRVtCwo4z9jPnQwTdxvm)". Not tested, since I can't get the server itself to launch properly.
> > > https://paste.ec/paste/dmAv-1yz#TDN6AFGOEARz4QKNjspLb6jgIzXeobvqk9EMEDOB248
> > 
> > 
> > You hit the same block I did it seems lol. It seems to hang with the SteamAPI check. Mine was using the sh file rather than the startup you have set [egg-humanit-z.json](https://github.com/parkervcp/eggs/files/13756281/egg-humanit-z.json)
> 
> What information would be needed to get past this roadblock? I can ask on their discord.

Just the logs from it starting to be honest? Its their files using the TSSServer sh file to start
```
steam user is not set.
Using anonymous user.
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/home/container/Steam/logs/stderr.txt'
Logging directory: '/home/container/Steam/logs'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation - version 1702079268
-- type 'quit' to exit --
Loading Steam API...dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory
OK
Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
Success! App '2728330' already up to date.
:/home/container$ ./TSSGameServer.sh -log -port=${SERVER_PORT} -queryport=${QUERY_PORT} -steamservername=${SERV_NAME}
4.27.2-0+++UE4+Release-4.27 522 0
Disabling core dumps.
Looking for binary: ../../../TSSGame/Config/BinaryConfig.ini
dlopen failed trying to load:
steamclient.so
with error:
steamclient.so: cannot open shared object file: No such file or directory
[S_API] SteamAPI_Init(): Loaded '/home/container/.steam/sdk64/steamclient.so' OK.  (First tried local 'steamclient.so')
CAppInfoCacheReadFromDiskThread took 8 milliseconds to initialize
dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory
Setting breakpad minidump AppID = 1766060
[S_API FAIL] Tried to access Steam interface SteamNetworking006 before SteamAPI_Init succeeded.
[S_API FAIL] Tried to access Steam interface SteamUser021 before SteamAPI_Init succeeded.
[S_API FAIL] Tried to access Steam interface SteamFriends017 before SteamAPI_Init succeeded.
```
Startup:
`./TSSGameServer.sh -log -port={{SERVER_PORT}} -queryport={{QUERY_PORT}} -steamservername={{SERV_NAME}}`

**Comment by QuintenQVD0** [comment]
Keep in mind that startup scripts are not allowed. So you will have to transfere the logic in the the startup cmd of the egg. So I would also like if you share any test egg here that it does not have a startup script as random users will come 1nd download it and use it.

**Comment by ImKringle** [comment]
> Keep in mind that startup scripts are not allowed. So you will have to transfere the logic in the the startup cmd of the egg. So I would also like if you share any test egg here that it does not have a startup script as random users will come 1nd download it and use it.

Is there any specific reason? 100% ok with me i'll remove the startup before sharing in the future; just curious as the devs are who shipped that option lol

**Comment by QuintenQVD0** [comment]
> > Keep in mind that startup scripts are not allowed. So you will have to transfere the logic in the the startup cmd of the egg. So I would also like if you share any test egg here that it does not have a startup script as random users will come 1nd download it and use it.
> 
> Is there any specific reason? 100% ok with me i'll remove the startup before sharing in the future; just curious as the devs are who shipped that option lol

They add an .sh file to make it easier for the end user so they do not have to search for the binary. Pterodactyl runs docker containers with some specific flags for security and isolation. That script (.sh) is user editable so that can just edit it to start miners, zipper boms and many more things. That is wy it is not allowed. as the egg, startup should not run a file so that the end user can not (at least easy) start something that should not be ran.

**Comment by ImKringle** [comment]
> > > Keep in mind that startup scripts are not allowed. So you will have to transfere the logic in the the startup cmd of the egg. So I would also like if you share any test egg here that it does not have a startup script as random users will come 1nd download it and use it.
> > 
> > 
> > Is there any specific reason? 100% ok with me i'll remove the startup before sharing in the future; just curious as the devs are who shipped that option lol
> 
> They add an .sh file to make it easier for the end user so they do not have to search for the binary. Pterodactyl runs docker containers with some specific flags for security and isolation. That script (.sh) is user editable so that can just edit it to start miners, zipper boms and many more things. That is wy it is not allowed. as the egg, startup should not run a file so that the end user can not (at least easy) start something that should not be ran.

Understood, makes sense

**Comment by vp-en** [comment]
Just a heads up.. I tried booting up a clean/empty VPS, Ubuntu 22.04LTS, to run SteamCMD natively.

The exact same thing happens, as it does in Pterodactyl/Docker.

`./TSSGameServer.sh -logs -port=7777 -queryport=27015 -steamservername=HMZ0012`

Results in the same thing:
```
4.27.2-0+++UE4+Release-4.27 522 0
Disabling core dumps.
Looking for binary: ../../../TSSGame/Config/BinaryConfig.ini
dlopen failed trying to load:
steamclient.so
with error:
steamclient.so: cannot open shared object file: No such file or directory
[S_API] SteamAPI_Init(): Loaded '/home/steam/.steam/sdk64/steamclient.so' OK.  (First tried local 'steamclient.so')
CAppInfoCacheReadFromDiskThread took 9 milliseconds to initialize
dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory
Setting breakpad minidump AppID = 1766060
[S_API FAIL] Tried to access Steam interface SteamNetworking006 before SteamAPI_Init succeeded.
[S_API FAIL] Tried to access Steam interface SteamUser021 before SteamAPI_Init succeeded.
[S_API FAIL] Tried to access Steam interface SteamFriends017 before SteamAPI_Init succeeded.
```

So there might be something up with either the instructions, or the server software itself. 

In the [eggs I previously linked](https://github.com/parkervcp/eggs/issues/2624#issuecomment-1867919840), I converted the `TSSGameServer.sh` script to this startup command:
`chmod +x ./TSSGame/Binaries/Linux/TSSGameServer-Linux-Shipping && ./TSSGame/Binaries/Linux/TSSGameServer-Linux-Shipping TSSGame "$@" -log -port={{SERVER_PORT}} -queryport={{QUERY_PORT}} -steamservername={{ServerName}}"`

But it produces the same result as mentioned in this post and above

**Comment by gOOvER** [comment]
in their Discord someone said, that some 32bit libs are missing. i dont find out, which libs

**Comment by QuintenQVD0** [comment]
> Just a heads up.. I tried booting up a clean/empty VPS, Ubuntu 22.04LTS, to run SteamCMD natively.
> 
> The exact same thing happens, as it does in Pterodactyl/Docker.
> 
> `./TSSGameServer.sh -logs -port=7777 -queryport=27015 -steamservername=HMZ0012`
> 
> Results in the same thing:
> 
> ```
> 4.27.2-0+++UE4+Release-4.27 522 0
> Disabling core dumps.
> Looking for binary: ../../../TSSGame/Config/BinaryConfig.ini
> dlopen failed trying to load:
> steamclient.so
> with error:
> steamclient.so: cannot open shared object file: No such file or directory
> [S_API] SteamAPI_Init(): Loaded '/home/steam/.steam/sdk64/steamclient.so' OK.  (First tried local 'steamclient.so')
> CAppInfoCacheReadFromDiskThread took 9 milliseconds to initialize
> dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory
> Setting breakpad minidump AppID = 1766060
> [S_API FAIL] Tried to access Steam interface SteamNetworking006 before SteamAPI_Init succeeded.
> [S_API FAIL] Tried to access Steam interface SteamUser021 before SteamAPI_Init succeeded.
> [S_API FAIL] Tried to access Steam interface SteamFriends017 before SteamAPI_Init succeeded.
> ```
> 
> So there might be something up with either the instructions, or the server software itself.
> 
> In the [eggs I previously linked](https://github.com/parkervcp/eggs/issues/2624#issuecomment-1867919840), I converted the `TSSGameServer.sh` script to this startup command: `chmod +x ./TSSGame/Binaries/Linux/TSSGameServer-Linux-Shipping && ./TSSGame/Binaries/Linux/TSSGameServer-Linux-Shipping TSSGame "$@" -log -port={{SERVER_PORT}} -queryport={{QUERY_PORT}} -steamservername={{ServerName}}`
> 
> But it produces the same result as mentioned in this post and above

the chmod +x is not needed, and if so it should only happen in the install script and the "$@" should be left out, and server name should be in "" so `-steamservername="{{ServerName}}"`

**Comment by vp-en** [comment]
> the chmod +x is not needed, and if so it should only happen in the install script and the "$@" should be left out, and server name should be in "" so `-steamservername="{{ServerName}}"`

Great, updated the eggs [in my post](https://github.com/parkervcp/eggs/issues/2624#issuecomment-1867919840) to reflect those changes. Still the same issue though, I think [gOOvER](https://github.com/parkervcp/eggs/issues/2624#issuecomment-1868312148) might be on to something about the missing libraries

**Comment by gOOvER** [comment]
> > the chmod +x is not needed, and if so it should only happen in the install script and the "$@" should be left out, and server name should be in "" so `-steamservername="{{ServerName}}"`
> 
> Great, updated the eggs [in my post](https://github.com/parkervcp/eggs/issues/2624#issuecomment-1867919840) to reflect those changes. Still the same issue though, I think [gOOvER](https://github.com/parkervcp/eggs/issues/2624#issuecomment-1868312148) might be on to something about the missing libraries

i open a Ticket, because the error indicate a Bug in the server and is not a problem on client side. I hope, i get an answer

**Comment by gOOvER** [comment]
I had contact with the lead Dev; he will take a look into it :)

**Comment by ImKringle** [comment]
![Screenshot 2023-12-24 120457](https://github.com/parkervcp/eggs/assets/135805091/0a1fff17-a908-4db8-83e3-4181d6920469)
New Update

**Comment by ImKringle** [comment]
![Screenshot 2023-12-24 120859](https://github.com/parkervcp/eggs/assets/135805091/db6dc727-00fa-43a2-8a33-d3cffacd5468)
Linux now boots

Will work on getting an egg together later unless @vp-en wants to hop on it as hes been working on this more

**Comment by ImKringle** [comment]
Basic egg here - Seems to boot but someone else will need to test connectivity, all I changed myself was the start configuration so it actually posts as running when loaded fully. If someone could test we can make a PR <3
[egg-humanit-z.json](https://github.com/parkervcp/eggs/files/13762428/egg-humanit-z.json)

**Comment by ValhallaCommunityOfficial** [comment]
> Basic egg here - Seems to boot but someone else will need to test connectivity, all I changed myself was the start configuration so it actually posts as running when loaded fully. If someone could test we can make a PR <3 [egg-humanit-z.json](https://github.com/parkervcp/eggs/files/13762428/egg-humanit-z.json)

![image](https://github.com/parkervcp/eggs/assets/142844832/28dbe6a4-fb8f-4fcd-9aaa-da1340008f9a)

Seems like it's working, although not seeing it on server list yet. using game port 7777 and query port 27015.

**Comment by ImKringle** [comment]
> > Basic egg here - Seems to boot but someone else will need to test connectivity, all I changed myself was the start configuration so it actually posts as running when loaded fully. If someone could test we can make a PR <3 [egg-humanit-z.json](https://github.com/parkervcp/eggs/files/13762428/egg-humanit-z.json)
> 
> ![image](https://private-user-images.githubusercontent.com/142844832/292683436-28dbe6a4-fb8f-4fcd-9aaa-da1340008f9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM0NDAwOTIsIm5iZiI6MTcwMzQzOTc5MiwicGF0aCI6Ii8xNDI4NDQ4MzIvMjkyNjgzNDM2LTI4ZGJlNmE0LWZiOGYtNGZjZC05YWFhLWRhMTM0MDAwOGY5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyNFQxNzQzMTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZGJlYzQ0OTEwY2U3ZTc4OTQ3ZTdiNGM5MzA1NzBiNDY4ZDI4ZTM2MGI2YzE2MmI4ZmI3MmM3ZTJjMDhmMTEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fJIgO2bFj4QqaTCIFxj90O0kKlE1A14OEujKKIyAN64)
> 
> Seems like it's working, although not seeing it on server list yet. using game port 7777 and query port 27015.

After checking it seems mine shows on Steam under the 27015 port
[
![Screenshot 2023-12-24 124437](https://github.com/parkervcp/eggs/assets/135805091/94ad2851-590e-47c3-8582-07ae7f7773f8)
](url)

Whatre you searching with?

**Comment by ValhallaCommunityOfficial** [comment]
Ah, well in that case I'm sure it works. I searched with Port 7777 by mistake. At the in-laws atm so i wouldn't be able to test further until tonight

**Comment by ValhallaCommunityOfficial** [comment]
> Basic egg here - Seems to boot but someone else will need to test connectivity, all I changed myself was the start configuration so it actually posts as running when loaded fully. If someone could test we can make a PR <3 [egg-humanit-z.json](https://github.com/parkervcp/eggs/files/13762428/egg-humanit-z.json)

Can confirm now that this egg works. Was able to join. No issues!

**Comment by vp-en** [comment]
Added a PR here: https://github.com/parkervcp/eggs/pull/2628

**Comment by ImKringle** [comment]
PR Is merged, this can be closed @ValhallaCommunityOfficial 🖖🏼

**Comment by QuintenQVD0** [comment]
Closed as of #2628
